rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,279 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# RFC Deletion and deprecation of data in the DCP
+
+
+## Summary
+Establish a secure and controlled process for deleting data from the Data Store.
+
+
+## Author(s)
+
+*Recommended format for Authors:*
+
+ `[Trent Smith](mailto:tsmith12@ucsc.edu)`
+ `[Hannes Schmidt](mailto:hannes@ucsc.edu)`
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+*Describe the user or technical need in detail [with alignment to the DCP roadmap priorities where possible]. Link prior community discussions to demonstrate support for this RFC.*
+
+One of the DCP's primary objectives is to ensure the durability of the
+(meta)data it contains and to maintain a complete record of the changes made to
+it. The DCP achieves this by never overwriting a (meta)data file but instead
+treating an update to such a file as the addition of a new version of that
+file. And instead of physically removing a file version, the DCP hides that
+file version behind a deletion marker.
+
+Under certain circumstances however, the *physical deletion* of a file version
+is required. The first section of this specification describes the
+circumstances and mechanisms for logical and physical deletion of files and the
+bundles that contain them.
+
+Independently, it may be required to discourage the use of (meta)data managed
+by the DCP, be it for replication, analysis or aggregation, without actually
+preventing that use. The DCP handles this by *deprecating* the (meta)data as
+specified in section 2.
+
+### User Stories
+
+*Share the [User Stories](https://www.mountaingoatsoftware.com/agile/user-stories) motivating this RFC.*
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+*Explain the design in sufficient detail such that the implementation and (if appropriate) the interaction with existing DCP software are both reasonably clear.*
+
+## Detailed Design
+
+### Reasons
+
+Reasons for deletion of (meta)data are
+
+* consent to share (meta)data derived from a donor's specimens is withdrawn by
+  that donor after the (meta)data was submitted to DCP
+  
+* that consent was never given but this fact is revealed only after the
+  (meta)data was submitted to the DCP
+  
+* storing or processing the (meta)data causes a serious disruption of service
+  within the DCP
```

Comment:
Valid data or metadata that causes our services to fail indicates a serious design flaw or bug. We should not need to resort to deletion in order to fix our system.

Reply:
but the flaw may not be from the DSS and it might easier to delete the files to avoid impeding the progress of others. Deletion in general should never happen. These are worst case scenarios.

Reply:
@kislyuk but you agree with the presented reason for deleting data?

Reply:
Yes.